int8  CMD_UP = 1
int8  CMD_DOWN = 3

int8  cmd